src/components/uielements/coins/coinData/CoinData.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+
+import { TokenAmount, formatTokenAmount } from '@thorchain/asgardex-token'
+import { bn, formatBN } from '@thorchain/asgardex-util'
+import BigNumber from 'bignumber.js'
+
+import Label from '../../label'
+import Coin from '../coin'
+import { CoinDataWrapper, CoinDataWrapperType, CoinDataWrapperSize } from './CoinData.style'
+
+type Props = {
+  'data-test'?: string
+  asset?: string
+  assetValue?: TokenAmount
+  target?: string
+  targetValue?: TokenAmount
+  price?: BigNumber
+  priceUnit?: string
+  priceValid?: boolean
+  size?: CoinDataWrapperSize
+  className?: string
+  type?: CoinDataWrapperType | undefined
+}
+
+const CoinData: React.FC<Props> = (props: Props): JSX.Element => {
+  const {
+    asset = 'bnb',
+    assetValue = null,
```

Comment:
[nitpick] No need to set default value for `assetValue` to `null`, since it's `undefined` by default using ?. Same for all other optional `xxxProps?`.